src/index/instance_resolver.rs

```diff
@@ -0,0 +1,151 @@
+use crate::{
```

Comment:
I think "indstance_iterator" better catches this file than 'instances_resolver' - are we resolving anything here?